custom_components/ha-google-home/api.py

```diff
@@ -52,62 +54,88 @@ def get_android_id(self):
+            url = "https://{ip}:{port}/{endpoint}".format(
+                ip=ip, port=str(port), endpoint=api_endpoint
+            )
+            return url
 
     def get_alarms_and_timers_from(self, device, endpoint):
         """Fetches timers and alarms from google device"""
-        url = self.create_url(device.ip, device.port, endpoint)
-        _LOGGER.debug(
-            "For device {device} - {url}".format(device=device.device_name, url=url)
-        )
-        # verify=False is need to avoid SSL security checks. Otherwise it will fail to connect"""
-        HEADERS[HEADER_CAST_LOCAL_AUTH] = device.local_auth_token
-        response = requests.get(url, headers=HEADERS, verify=False, timeout=TIMEOUT)
-
-        if response.status_code != HTTP_OK:
-            _LOGGER.error(
-                "For device {device} - API returned {error}: {reason}".format(
-                    device=device.device_name,
-                    error=response.status_code,
-                    reason=response.text,
-                )
+        url = self.create_url(device.ip, PORT, endpoint)
+        if url:
+            _LOGGER.debug(
+                "For device {device} - {url}".format(device=device.device_name, url=url)
             )
-            return
+            # verify=False is need to avoid SSL security checks. Otherwise it will fail to connect"""
+            HEADERS[HEADER_CAST_LOCAL_AUTH] = device.local_auth_token
+            response = requests.get(url, headers=HEADERS, verify=False, timeout=TIMEOUT)
+
+            if response.status_code != HTTP_OK:
+                _LOGGER.error(
+                    "For device {device} - API returned {error}: {reason}".format(
+                        device=device.device_name,
+                        error=response.status_code,
+                        reason=response.text,
+                    )
+                )
+                return
+            else:
+                return response.json()
         else:
-            return response.json()
+            return
 
     def get_google_devices_information(self):
         """Retrieves devices from glocaltokens"""
         _LOGGER.debug("Fetching data...")
+        offline_devices = []
         devices = self._client.get_google_devices()
 
         for device in devices:
             # To avoid key error's
             timers = []
             alarms = []
-
-            result = self.get_alarms_and_timers_from(device, API_ENDPOINT_ALARMS)
-            if result:
-                timers = result.get(TIMERS)
-                alarms = result.get(ALARMS)
-
-                if not timers and not alarms:
-                    _LOGGER.error(
-                        "For device {device} - {error}".format(
-                            device=device.device_name, error=API_RETURNED_UNKNOWN
+            if device.ip:
+                result = self.get_alarms_and_timers_from(device, API_ENDPOINT_ALARMS)
+                if result:
```

Comment:
What if this returns nothing or a wrong data?
I suggest remove this check, because the next check `if not timers and not alarms` will cater for it and log error

Reply:
I'm not use i understand what you are referering to?

Reply:
Sorry, i was unclear. I was confused what if `result` is nothing. Then we just skip the device and do nothing. I suggest we remove `if result:` at all and rely on later check `if not timers and not alarms`

Reply:
`result` will return `False` if a http error occurred, and `if not timers and not alarms` is just to check if the JSON returned includes the necessary attributes.

`if not timers and not alarms` error should only happen if Google changes the API

Reply:
oh, you are right, sorry, forgot that `results` is a json response. Maybe we could also rename it to `result_json` for better clarity? 😄 Will just mark it resolved then 👍

Reply:
Good idea :smile: